### PR TITLE
remove reference to GoQuorum-compatible privacy

### LINKED ISSUE #1218: [Content] Deprecate GoQuorum-compatible privacy in Besu
### Describe the issue

Reference to this feature can be removed from Besu docs

### What page has the issue?

https://besu.hyperledger.org/en/stable/private-networks/how-to/use-privacy/goquorum-compatible

### Issue type

Outdated content

### Change suggestion

also referenced within Tessera content

### More context

_No response_

### Besu documentation issue guidelines

- [X] I read and followed this project's documentation guidelines.

### PR DESCRIPTION
**Pull request by macfarla**
Signed-off-by: Sally MacFarlane <macfarla.github@gmail.com>

This feature is to be deprecated in besu 23.1

Make sure that:

- [x] [All commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] You've read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] You've [previewed your changes locally](https://wiki.hyperledger.org/display/BESU/Preview+the+documentation).

## Describe the change

Remove reference to GoQuorum-compatible privacy

Fixes #1218 

For content changes:

- [x] Documentation content
- [x] Documentation page organization

## After creating your PR and tests have finished

Make sure that:

- [ ] You've fixed any issues raised by the tests.
- [x] You've [previewed your changes on Read the Docs](https://wiki.hyperledger.org/display/BESU/Preview+the+documentation)
  and added a [preview link](#preview).

## Preview
https://hyperledger-besu--1219.org.readthedocs.build/en/1219/private-networks/concepts/privacy/

### CODE REVIEW

File: docs/private-networks/tutorials/privacy/multi-tenancy.md [hunk 1]
```diff
@@ -132,9 +132,8 @@ In the `Node-1/Tessera` directory, update the `tessera.conf` file by adding the
 
 !!! note
 
-    If you are running Besu in [GoQuorum-compatible privacy mode](../../how-to/use-privacy/goquorum-compatible.md),
-    disable [`orion` mode](https://docs.tessera.consensys.net/en/stable/HowTo/Configure/Orion-Mode/) by removing the line
-    `"mode": "orion",` from the Tessera configuration file.
+    Besu requires [`orion` mode](https://docs.tessera.consensys.net/en/stable/HowTo/Configure/Orion-Mode/) add the line
+    `"mode": "orion",` to the Tessera configuration file.
```
Comment (rolandtyler) [inline docs/private-networks/tutorials/privacy/multi-tenancy.md:136]:
```suggestion
    Besu requires [`orion` mode](https://docs.tessera.consensys.net/en/stable/HowTo/Configure/Orion-Mode/). Add the line
    `"mode": "orion",` to the Tessera configuration file.
```
I think we can remove this note altogether, though. It's already in the config example in the file, and we don't call it out in the other examples.

**Review by rolandtyler — APPROVED** [review]
Minor nitpick, but LGTM. Was this feature widely used enough to warrant a deprecation notice anywhere?

**Comment by macfarla** [comment]
> Minor nitpick, but LGTM. Was this feature widely used enough to warrant a deprecation notice anywhere?

it was not used which is the main reason it's being deprecated.

File: mkdocs.redirects.yml [hunk 2]
```diff
@@ -126,7 +126,6 @@ plugins:
       Reference/Resources.md: index.md
       HowTo/Use-Privacy/EEA-compliant.md: private-networks/how-to/use-privacy/eea-compliant.md
       HowTo/Use-Privacy/Privacy.md: private-networks/how-to/use-privacy/besu-extended.md
-      HowTo/Use-Privacy/Use-GoQuorum-compatible-privacy.md: private-networks/how-to/use-privacy/goquorum-compatible.md
       HowTo/Use-Privacy/Run-Tessera-With-Besu.md: private-networks/how-to/use-privacy/tessera.md
```
Comment (alexandratran) [inline mkdocs.redirects.yml:129]:
```suggestion
      HowTo/Use-Privacy/Use-GoQuorum-compatible-privacy.md: private-networks/concepts/privacy/index.md
      private-networks/how-to/use-privacy/goquorum-compatible.md: private-networks/concepts/privacy/index.md
      HowTo/Use-Privacy/Run-Tessera-With-Besu.md: private-networks/how-to/use-privacy/tessera.md
```

Deleting a page requires adding a redirect - I just redirected to the main privacy concepts page for now. Deleted redirects also must be replaced.

**Review by NicolasMassart — APPROVED** [review]
LGTM.
I disabled Gitleak on Besu doc as it has no clear added value. This check was recently added in super-linter which is a meta linter including multiple linters. So we inherit their additions and we do not necessarily need them. I first tried to exclude keys, but it looks like it's not suitable once we start making changes in the pages.
Also, I think the need to check secrets in doc is not as important as in code repo where it could be left in a hardcoded variable after testing or initial dev. But here, only places are code sample and we only use test values that can go public.

**Pull request merged**